pinot-core/src/test/java/com/linkedin/pinot/core/plan/maker/PlanNodeFactoryTest.java

```diff
@@ -0,0 +1,322 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.plan.maker;
+
+import java.io.File;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.io.FileUtils;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import com.linkedin.pinot.common.data.FieldSpec.DataType;
+import com.linkedin.pinot.common.data.Schema;
+import com.linkedin.pinot.common.data.FieldSpec;
+import com.linkedin.pinot.common.data.StarTreeIndexSpec;
+import com.linkedin.pinot.common.request.BrokerRequest;
+import com.linkedin.pinot.common.segment.ReadMode;
+import com.linkedin.pinot.core.indexsegment.IndexSegment;
+import com.linkedin.pinot.core.indexsegment.columnar.ColumnarSegmentLoader;
+import com.linkedin.pinot.core.indexsegment.generator.SegmentGeneratorConfig;
+import com.linkedin.pinot.core.plan.AggregationGroupByPlanNode;
+import com.linkedin.pinot.core.plan.AggregationPlanNode;
+import com.linkedin.pinot.core.plan.DictionaryBasedAggregationPlanNode;
+import com.linkedin.pinot.core.plan.MetadataBasedAggregationPlanNode;
+import com.linkedin.pinot.core.plan.PlanNode;
+import com.linkedin.pinot.core.plan.SelectionPlanNode;
+import com.linkedin.pinot.core.segment.creator.SegmentIndexCreationDriver;
+import com.linkedin.pinot.core.segment.creator.impl.SegmentIndexCreationDriverImpl;
+import com.linkedin.pinot.pql.parsers.Pql2Compiler;
+
+
+public class PlanNodeFactoryTest {
```

Comment:
very good test case but this is not really PlanNodeFactoryTest, this is testing specifically for the min/max/minmax case where we can use metadata/dictionary to execute the the query. We should rename the test case to reflect that.